### ISSUE TITLE
weapon powerup can boost charge too high for animation (indexOutOfBounds)

### ISSUE THREAD

**Issue by 7yl4r** [EmergentOrganization/cell-rpg#113]
```
Exception in thread "LWJGL Application" java.lang.IndexOutOfBoundsException: Index: 10, Size: 5
    at java.util.ArrayList.rangeCheck(ArrayList.java:635)
    at java.util.ArrayList.get(ArrayList.java:411)
    at java.util.Collections$UnmodifiableList.get(Collections.java:1211)
    at io.github.emergentorganization.cellrpg.core.entityfactory.Entities.Equipment.ChargeAnimatedEquipment.getChargeFrame(ChargeAnimatedEquipment.java:61)
    at io.github.emergentorganization.cellrpg.core.entityfactory.Entities.Equipment.ChargeAnimatedEquipment.getChargeFrame(ChargeAnimatedEquipment.java:54)
    at io.github.emergentorganization.cellrpg.core.entityfactory.Entities.Equipment.ChargeAnimatedEquipment.onChargeChanged(ChargeAnimatedEquipment.java:50)
    at io.github.emergentorganization.cellrpg.core.entityfactory.Entities.Equipment.Weapon.recharge(Weapon.java:71)
    at io.github.emergentorganization.cellrpg.components.EquipmentList.rechargeEquipment(EquipmentList.java:27)
    at io.github.emergentorganization.cellrpg.systems.EnergySystem.process(EnergySystem.java:35)
    at com.artemis.systems.IntervalEntityProcessingSystem.processEntities(IntervalEntityProcessingSystem.java:48)
```

**Comment by 7yl4r** [comment]
This could be fixed by improving handling of `Equipment.charge` . It should be made private and setter should trigger `onChargeChange`.

**Comment by 7yl4r** [comment]
closed by 1836912
